src/Bicep.Core/TypeSystem/Providers/ThirdParty/ThirdPartyResourceTypeLoader.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Azure.Bicep.Types;
+using Azure.Bicep.Types.K8s;
+using Bicep.Core.Resources;
+using Bicep.Core.TypeSystem.Providers.Az;
+using Bicep.Core.TypeSystem.Providers.K8s;
+using Bicep.Core.TypeSystem.Types;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Bicep.Core.TypeSystem.Providers.ThirdParty
+{
+    public class ThirdPartyResourceTypeLoader : IResourceTypeLoader
+    {
+        private readonly ITypeLoader typeLoader;
+        //No need to create a 3rd party factory
+        private readonly K8sResourceTypeFactory resourceTypeFactory;
+        private readonly ImmutableDictionary<ResourceTypeReference, TypeLocation> availableTypes;
+        private readonly ImmutableDictionary<string, ImmutableDictionary<string, ImmutableArray<TypeLocation>>> availableFunctions;
+
+        public ThirdPartyResourceTypeLoader(ITypeLoader typeLoader)
+        {
+            this.typeLoader = typeLoader;
+            resourceTypeFactory = new K8sResourceTypeFactory();
+            var indexedTypes = typeLoader.LoadTypeIndex();
+            availableTypes = indexedTypes.Resources.ToImmutableDictionary(
+                kvp => ResourceTypeReference.Parse(kvp.Key),
+                kvp => kvp.Value);
+            //Harsh - do we need functions? what are they? 
```

Comment:
Not currently, but we may want this capability in the future. Functions allow the user to type e.g.:
```bicep
// implicitly loaded from the "az" provider
var myRg = resourceGroup()

// explicitly loaded from the "az" provider
var myRg = az.resourceGroup()
```